### PR TITLE
Corrected DNN elementwise multiplication

### PR DESCRIPTION
**Pull request by LupusSanctus**
Merge with extra: opencv/opencv_extra#865
Fixes [#18214](https://github.com/opencv/opencv/issues/18214#issuecomment-683894976): 
* issue 1:  aligned OCV and [TF multiplication](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Multiply?hl=ru) layer behaviour for the following case: inputs **a** (shape: [1,1,1, C] ) and **b** (shape: [1, H, W, C] ), OCV **a** mul **b** returns [1,1,1, C], whereas TF mul - [1, H, W, C].
* issue 2: MobileNetV3 reshape problem (before Conv layer): ``number of input channels should be multiple of 72 but got 1 in getMemoryShapes()``

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2021.2.0:20.04
build_image:Custom Win=openvino-2021.1.0
build_image:Custom Mac=openvino-2021.2.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*

allow_multiple_commits=1
```

### CODE REVIEW

**Comment by alalek** [comment]
@LupusSanctus Please rebase commits (conflict with merged #19477)

**Comment by LupusSanctus** [comment]
@alalek, rebase was done.

**Review by alalek — APPROVED** [review]
Thank you!
